### PR TITLE
[Keyboard] Add Hi-Tek AIM-65 keyboard

### PR DESCRIPTION
**Pull request by Bonnee**
## Description
The keboard assembly that came with the Rockwell AIM-65, a microcomputer from the '70s.
I reverse-engineered the matrix and wired a Pro Micro to the original connector.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/handwired/aim65/rules.mk [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+#
+BOOTMAGIC_ENABLE ?= yes	# Virtual DIP switch configuration(+1000)
+MOUSEKEY_ENABLE ?= no	# Mouse keys(+4700)
+EXTRAKEY_ENABLE ?= yes	# Audio control and System control(+450)
+CONSOLE_ENABLE ?= no	# Console for debug(+400)
+COMMAND_ENABLE ?= no    # Commands for debug and configuration
+SLEEP_LED_ENABLE ?= no  # Breathing sleep LED during USB suspend
+NKRO_ENABLE ?= no	# USB Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+BACKLIGHT_ENABLE ?= no  # Enable keyboard backlight functionality
+AUDIO_ENABLE ?= no
```
Comment (zvecr) [inline keyboards/handwired/aim65/rules.mk:17]:
```suggestion
# Bootloader selection
BOOTLOADER = caterina

# Build Options
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = no            # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
```

File: keyboards/handwired/aim65/readme.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+The controller is a Pro Micro handwired to the original connector. [This](https://github.com/Bonnee/aim-65_keyboard) repo collects 3D models for the case, schematics for the matrix and pics.
+
+* Keyboard Maintainer: [Matteo Bonora](https://github.com/bonnee)
+* Hardware Supported: Pro Micro
+
+The default layout is useless on a modern computer, you can use it as a base for your own though.
+
+Make example for this keyboard (after setting up your build environment):
+
+    make aim_65:bonnee
```
Comment (zvecr) [inline keyboards/handwired/aim65/readme.md:16]:
```suggestion
    make handwired/aim65:default
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/handwired/aim65/readme.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
+# Hi-Tek AIM-65
+
+![Hi-Tek AIM-65 Keyboard](https://imgur.com/7Gam1f2.jpg)
```
Comment (fauxpark) [inline keyboards/handwired/aim65/readme.md:3]:
```suggestion
![Hi-Tek AIM-65 Keyboard](https://imgur.com/7Gam1f2l.jpg)
```

File: keyboards/handwired/aim65/rules.mk [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
+AUDIO_ENABLE = no           # Audio output
+RGBLIGHT_ENABLE ?= no
```
Comment (fauxpark) [inline keyboards/handwired/aim65/rules.mk:23]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/handwired/aim65/keymaps/bonnee/keymap.c [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include QMK_KEYBOARD_H
+
+#define CTESC LCTL_T(KC_ESC)
+
+enum layer {
+    BL = 0,
+    FL = 1
```
Comment (drashna) [inline keyboards/handwired/aim65/keymaps/bonnee/keymap.c:23]: The numbering isn't needed, actually. It starts at 0, and automatically increments.
```suggestion
    BL,
    FL
```

File: keyboards/handwired/aim65/keymaps/default/keymap.c [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    LAYOUT(
+	     KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0, KC_SCLN,   KC_F3, KC_PSCR,
```
Comment (drashna) [inline keyboards/handwired/aim65/keymaps/default/keymap.c:21]:
```suggestion
         KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0, KC_SCLN,   KC_F3, KC_PSCR,
```

File: keyboards/handwired/aim65/keymaps/default/keymap.c [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+      KC_ESC,    KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P, KC_MINS,   KC_F2,KC_RETURN,
+      KC_LCTRL,    KC_A, KC_S, KC_D, KC_F, KC_G, KC_H, KC_J, KC_K, KC_L, KC_SCLN, KC_AT, KC_F1, KC_DEL,
+           KC_LSFT,    KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M, KC_COMM,  KC_DOT, KC_SLSH, KC_RSFT,
+							      KC_SPC
```
Comment (drashna) [inline keyboards/handwired/aim65/keymaps/default/keymap.c:25]:
```suggestion
                                  KC_SPC
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/handwired/aim65/keymaps/bonnee/keymap.c (hunk 5 above)
Comment (Bonnee) [reply keyboards/handwired/aim65/keymaps/bonnee/keymap.c]: Right! numbering was from when I used defines

File: keyboards/handwired/aim65/keymaps/bonnee/keymap.c [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include QMK_KEYBOARD_H
+
+#define CTESC LCTL_T(KC_ESC)
+
+enum layer {
```
Comment (fauxpark) [inline keyboards/handwired/aim65/keymaps/bonnee/keymap.c:21]:
```suggestion
enum layer_names {
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/handwired/aim65/config.h [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0xFEED
+#define PRODUCT_ID      0x6060
```
Comment (noroadsleft) [inline keyboards/handwired/aim65/config.h:23]:
```suggestion
#define PRODUCT_ID      0x0F34
```

`echo -n "AIM-65" | shasum | cut -c1-4 | echo "0x$(tr '[:lower:]' '[:upper:]')"`

**Review by noroadsleft — CHANGES_REQUESTED** [review]

**Review by noroadsleft — APPROVED** [review]
Looks good to me.

**Review by fauxpark — APPROVED** [review]

**Review by drashna — APPROVED** [review]
Thanks!

**Pull request merged**